### PR TITLE
DOC: Extend docstring pandas core index to_frame method

### PR DESCRIPTION
**Pull request by stijnvanhoey**
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

This PR provides adaptations of the original `to_frame` docstring, taking into account the [docstring guide](https://python-sprints.github.io/pandas/guide/pandas_docstring.html).

### CODE REVIEW

File: pandas/core/indexes/base.py [hunk 1]
```diff
@@ -1110,19 +1110,37 @@ def to_series(self, index=None, name=None):
...
-        Create a DataFrame with a column containing the Index.
+        """Create a DataFrame with a column containing the Index.
 
         .. versionadded:: 0.21.0
 
         Parameters
         ----------
-        index : boolean, default True
-            Set the index of the returned DataFrame as the original Index.
+        index : boolean
```
Comment (jreback) [inline pandas/core/indexes/base.py]: @datapythonista @jorisvandenbossche defaults on the same line, yes?

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/indexes/base.py (hunk 1 above)
Comment (datapythonista) [reply pandas/core/indexes/base.py]: Yes, the documentation still says to use it the numpy way (in brackets after the description), but in pandas-dev the consensus seems to be keeping the pandas way, which was already right in this case. Will update the documentation asap and send an email with the last minute changes to the organizers

Comment (jorisvandenbossche) [reply pandas/core/indexes/base.py]: Yes, the docstring guide has been updated, but apparently only in the source: https://github.com/python-sprints/python-sprints.github.io/blob/master/pandas/guide/source/pandas_docstring.rst and not yet in the online version

**Comment by stijnvanhoey** [comment]
There is indeed a difference between the online version and the `rst` source. I used the online guide to do the adaptations. Fixed now

File: pandas/core/indexes/base.py [hunk 2]
```diff
@@ -1122,7 +1122,25 @@ def to_frame(self, index=True):
 
         Returns
         -------
-        DataFrame : a DataFrame containing the original Index data.
+        DataFrame
+            DataFrame containing the original Index data.
+
+        Examples
+        --------
+        >>> idx = pd.Index(['Ant', 'Bear', 'Cow'])
```
Comment (datapythonista) [inline pandas/core/indexes/base.py]: Minor thing, but it'd probably look a bit nicer if you create the index with `name='animal'`. :) For the rest looks great to me.

File: pandas/core/indexes/base.py [hunk 3]
```diff
@@ -1110,13 +1110,14 @@ def to_series(self, index=None, name=None):
...
-        """Create a DataFrame with a column containing the Index.
+        """
+        Create a DataFrame with a column containing the Index.
 
         .. versionadded:: 0.21.0
 
         Parameters
         ----------
-        index : boolean (default True)
+        index : boolean, default True
```
Comment (stijnvanhoey) [inline pandas/core/indexes/base.py]: @jorisvandenbossche on the [docstring guide rst file](https://github.com/python-sprints/python-sprints.github.io/blob/master/pandas/guide/source/pandas_docstring.rst), the example dosctring provides: `color : str (default 'blue')`. So should it be `(default True)` or `, default True`?

Comment (jorisvandenbossche) [reply pandas/core/indexes/base.py]: Whoops, I was using the virtual in my head :-) That's something we need to fix, we decided on `, default True` I think

File: pandas/core/indexes/base.py (hunk 2 above)
Comment (stijnvanhoey) [reply pandas/core/indexes/base.py]: Well, I do think it is interesting when creating a new index:
```
>>> idx.to_frame(index = False)
  animal
0    Ant
1   Bear
2    Cow
```
but when using the same as index and column, it does not get 'nicer':

```
>>> idx.to_frame(index = True)
       animal
animal       
Ant       Ant
Bear     Bear
Cow       Cow
```

File: pandas/core/indexes/base.py (hunk 3 above)
Comment (stijnvanhoey) [reply pandas/core/indexes/base.py]: See, https://github.com/python-sprints/python-sprints.github.io/pull/62

**Pull request merged**